darts/utils/data/training_dataset.py

```diff
@@ -110,9 +113,184 @@ def _memory_indexer(
+                start_time = target_time_index[future_start]
+                end_time = target_time_index[future_end - 1]
+
+                if (
+                    start_time not in sample_weight_time_index
+                    or end_time not in sample_weight_time_index
+                ):
+                    raise_log(
+                        ValueError(
+                            f"Missing sample weights; could not find sample weights in index "
+                            f"value range: {start_time} - {end_time}."
+                        ),
+                        logger=logger,
+                    )
+
+                sample_weight_start = sample_weight_time_index.get_loc(start_time)
+                sample_weight_end = sample_weight_time_index.get_loc(end_time) + 1
+
+            # store position of initial sample and all relevant sub set indices
+            self._index_memory[target_idx] = {
+                "end_of_output_idx": end_of_output_idx,
+                "past_target": (past_start, past_end),
+                "future_target": (future_start, future_end),
+                "covariate": (covariate_start, covariate_end),
+                "sample_weight": (sample_weight_start, sample_weight_end),
+            }
+        else:
+            # load position of initial sample and its sub set indices
+            end_of_output_idx_last = self._index_memory[target_idx]["end_of_output_idx"]
+            past_start, past_end = self._index_memory[target_idx]["past_target"]
+            future_start, future_end = self._index_memory[target_idx]["future_target"]
+            covariate_start, covariate_end = self._index_memory[target_idx]["covariate"]
+            sample_weight_start, sample_weight_end = self._index_memory[target_idx][
+                "sample_weight"
+            ]
+
+            # evaluate how much the new sample needs to be shifted, and shift all indexes
+            idx_shift = end_of_output_idx - end_of_output_idx_last
+            past_start += idx_shift
+            past_end += idx_shift
+            future_start += idx_shift
+            future_end += idx_shift
+            covariate_start = (
+                covariate_start + idx_shift if covariate_start is not None else None
+            )
+            covariate_end = (
+                covariate_end + idx_shift if covariate_end is not None else None
+            )
+            sample_weight_start = (
+                sample_weight_start + idx_shift
+                if sample_weight_start is not None
+                else None
+            )
+            sample_weight_end = (
+                sample_weight_end + idx_shift if sample_weight_end is not None else None
+            )
+
+        return (
+            past_start,
+            past_end,
+            future_start,
+            future_end,
+            covariate_start,
+            covariate_end,
+            sample_weight_start,
+            sample_weight_end,
+        )
+
+    def _memory_indexer2(
```

Comment:
What is the difference between this method and `_memory_indexer` above?

Reply:
forgot to remove this method, thanks